clients/upslog.c

```diff
@@ -407,14 +443,28 @@ int main(int argc, char **argv)
 
 	printf("Network UPS Tools %s %s\n", prog, UPS_VERSION);
 
-	while ((i = getopt(argc, argv, "+hs:l:i:f:u:Vp:FB")) != -1) {
+	while ((i = getopt(argc, argv, "+hs:l:i:f:u:Vp:FBm:")) != -1) {
 		switch(i) {
 			case 'h':
 				help(prog);
 #ifndef HAVE___ATTRIBUTE__NORETURN
 				break;
 #endif
 
+			case 'm':
+				monhost_child_prev = monhost_child_current;
+				monhost_child_current = xmalloc(sizeof(struct monhost_child));
+				if (monhost_child_anchor == NULL)
+					monhost_child_anchor = monhost_child_current;
+				else
+					monhost_child_prev->next = monhost_child_current;
+				monhost_child_current->next = NULL;
+				m_arg = optarg;
+				monhost_child_current->monhost = strsep(&m_arg, ",");
+				monhost_child_current->logfn = strsep(&m_arg, ",");
+				monhost_child_current->pidfilebase = strsep(&m_arg, ",");
```

Comment:
@cschuber : I think sanity-check is missing here, that these 3 strings are non-trivial.

Also, do I remember right that `strsep` would change the original string (`optarg`) and return pointers to its pieces? 
* Do we risk that original `optarg` would be released from RAM (e.g. is a temporary copy of `argv[X]`) on some systems?
* If it is just a pointer to `argv[X]`, are we okay to mangle original arguments array?

I think it would be more solid on all these fronts to make a copy of optarg, mangle it, `xstrdup` the pieces, free the copy.

Reply:
Note: addressed this idea in a later PR.

Reply:
I don't think it should matter but for the sake of correctness marg = optarg could be replaced by xstrdup(). It's the correct thing to do.

BTW, argv is only freed on execv() but using xstrdup is the better approach.